### ISSUE TITLE
Weird behaviors when using swc jest

### ISSUE THREAD

**Issue by mdbetancourt** [wallabyjs/public#2724]
### Issue description or question
with automatic configuration swc jest have weird behaviors like is not detecting changes properly or console.logs are out of lines

**Comment by smcenlly** [comment]
The problem that you describe is commonly caused by sourcemaps not being emitted for transpiled code. We are not familiar with `swc` but after reading their docs, it looks like your problem may be that source maps are not enabled. Can you confirm that you have [source maps set to true](https://swc.rs/docs/configuring-swc#sourcemaps)?

If you're still having problems, could you please provide us with a sample project that has `swc` + `jest` configured the same way as your project. If you are using TypeScript, please include any `tsconfig.json` settings and also any relevant `swc` settings for running TypeScript.

**Comment by smcenlly** [comment]
We set up and configured `swc` and `jest` with source maps enabled and also encountered some strange behaviors.

Unfortunately there appear to be a number of issues with `swc` source maps; they appear to missing some information, which means that Wallaby and other tools that require source maps will not operate correctly.

We have updated Wallaby to better handle some of these cases but there are still problems that we cannot work around or fix. Wallaby core `v1.0.1091` provides better support for `swc`.

We started by trying to configure one of our existing TypeScript projects to use `swc` instead of `babel-typescript` and immediately ran into [this known `swc` issue](https://github.com/swc-project/swc/issues/1685). Depending on your project configuration (e.g. for a simple JavaScript project), you may find that Wallaby mostly works for you with `swc`.

If you run into problems, please provide us with a stand-alone code sample and steps to reproduce; we can then identify the cause of the problems and if they're bugs in `swc`, we can help you report those to the `swc` team. 

We also highly recommend attempting to reproduce any problems outside of Wallaby where you can. For example, you can use `jest --collectCoverage` to identify where line coverage doesn't align with original source. You may also try `throw new Error()` and identifying where your error stack traces do not align correctly with your source code. This will allow you to directly report the issue to the `swc` team and will reduce turn-around times. We were able to quickly/easily see the same incorrect line mapping problems in stack traces, etc. when using jest without Wallaby.

**Comment by smcenlly** [comment]
FYI - we have added a new issue to `swc` repository with other known problems that cause Wallaby to not operate properly. [swc-project/swc#1796 - Source maps missing end range](https://github.com/swc-project/swc/issues/1796).

**Comment by mdbetancourt** [comment]
Hello thank you very much for you fast reply and your amazing job, looks like kdy1 added it to a milestone https://github.com/swc-project/swc/milestone/164 :+1:

**Comment by mdbetancourt** [comment]
Hello guys, and reopening this issue because kdy1 already published the fix to sourcemaps but the weird behavior persists, currently wallaby only works properly when is started at the first time, maybe wallaby require any extra configuration that im missing in order to work properly?

**Comment by ArtemGovorov** [comment]
@mdbetancourt It's almost certain that the issue is still with swc (especially given that other transformers are working properly). Can you please create a sample repo demonstrating the issue? We'll take a look to see what's exactly wrong to possibly raise an issue in swc repo.

**Comment by smcenlly** [comment]
We've created a reproduceable sample here: https://github.com/wallabyjs/wallaby-2724

As @ArtemGovorov said, we do not believe that this is an issue with Wallaby but we will confirm. It seems like something in `swc` is returning different results when the transform is called multiple times. You may force Wallaby to restart between test executions using the `workers.restart` flag, which appears to fix the problem. For example, update your `package.json` to:

**package.json**
```javascript
...
  "wallaby": {
    "workers": {
      "restart": true
    }
  }
}
```

**Comment by smcenlly** [comment]
We have identified that the subsequent test execution problem was a bug in Wallaby. This has been fixed in Wallaby core `v1.0.1100`.

We also tested the latest version of `swc` against one of our own more complex `TypeScript` / `Jest` projects. We had some tests failing after switching (that were passing prior to switching to `swc`) but then updated our `target` to be `es2018` and everything worked for us. In case you are interested in our `.swcrc` config, it was:

```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": false,
      "decorators": false,
      "dynamicImport": true
    },
    "target": "es2018"
  },
  "sourceMaps": true
}
```
